packages/itwinui-react/src/core/Tree/Tree.tsx

```diff
@@ -374,34 +398,57 @@ const VirtualizedTree = React.forwardRef(
       flatNodesList,
       itemRenderer,
       scrollToIndex,
-      className,
       style,
       ...rest
     }: VirtualizedTreeProps<T>,
-    ref: React.ForwardedRef<HTMLUListElement>,
+    ref: React.ForwardedRef<HTMLDivElement>,
   ) => {
-    const { outerProps, innerProps, visibleChildren } = useVirtualization({
-      itemsLength: flatNodesList.length,
-      itemRenderer: itemRenderer,
-      scrollToIndex,
+    const parentRef = React.useRef<HTMLDivElement | null>(null);
+
+    const getItemKey = React.useCallback(
+      (index: number) => {
+        return flatNodesList[index].nodeProps.nodeId;
+      },
+      [flatNodesList],
+    );
+
+    const virtualizer = useVirtualScroll({
+      count: flatNodesList.length,
+      getScrollElement: () => parentRef.current,
+      estimateSize: () => 39, //Set to 39px since that is the height of a treeNode with a sub label with the default font size.
+      getItemKey,
     });
 
+    useLayoutEffect(() => {
+      if (scrollToIndex) {
+        virtualizer.scrollToIndex(scrollToIndex);
+      }
+    }, [virtualizer, scrollToIndex]);
+
     return (
-      <Box
-        {...{
-          ...outerProps,
-          className: cx(className, outerProps.className),
-          style: { ...style, ...outerProps.style },
-        }}
+      <TreeElement
+        style={{ minInlineSize: '100%', overflow: 'auto', ...style }}
```

Comment:
The `minInlineSize: 100%` was left over from adapting the old virtualization implementation. After some testing, it seems that removing it causes no visual changes, so I have removed it in this [commit](https://github.com/iTwin/iTwinUI/pull/2091/commits/30832c871fa95f4997753532da6251953d8be638).

I did consider having users set `overflow: auto`, but I thought it would be strange for anyone newly adopting the virtualized `Tree` to have one virtualized component in our framework that asks users to set `overflow: auto`. The current implementation should work fine if users move the same styles they had before onto the `Tree` component, since the `overflow: auto` will just overwrite the `overflow: auto` that we are setting anyways.

Reply:
Yeah that makes sense. Should we add `overflow: auto` in the CSS then? And have it apply regardless of virtualization? Because that's what we do in Table.

Reply:
That makes sense. Added `overflow: auto` to the `.iui-tree` class and changesets for this change to both css and react in this [commit](https://github.com/iTwin/iTwinUI/pull/2091/commits/bc5fb3d42571d93ba8c9c774c49fb86ee11986ac).

Reply:
nit: It can be moved up before the nested block (after line 17) for better readability

![](https://github.com/iTwin/iTwinUI/assets/9084735/1d42a41c-9ffb-45f9-b05b-8fae0855d0b6)

Reply:
Moved in this [commit](https://github.com/iTwin/iTwinUI/pull/2091/commits/2ab621bd816341310ffba7ac1318330ee6bd3abb).